conda/cli/main_search.py

```diff
@@ -25,6 +27,29 @@ def execute(args, parser):
     else:
         subdirs = context.subdirs
 
+    if args.envs:
+        with spinner("Searching environments for %s" % spec,
+                     not context.verbosity and not context.quiet,
+                     context.json):
+            prefix_matches = search_all_prefixes(spec)
+        formatted_result = tuple({
+            'location': prefix,
+            'package_refs': tuple(PackageRef.from_objects(prefix_rec)
+                                  for prefix_rec in prefix_recs),
+        } for prefix, prefix_recs in iteritems(prefix_matches))
+        if context.json:
+            stdout_json(formatted_result)
+        else:
+            builder = []
+            for pkg_group in formatted_result:
+                builder.append("location: %s" % pkg_group['location'])
+                builder.append("package matches:%s" % dashlist(
+                    pref.dist_str() for pref in pkg_group['package_refs']
```

Comment:
The outputs with and without `--envs` are visually very different:
```
$ conda search defaults::python=3.6.4=hc3d631a_1
Loading channels: done
Name                       Version                   Build  Channel
python                     3.6.4                hc3d631a_1  defaults
```
```
$ conda search defaults::python=3.6.4=hc3d631a_1 --envs
Searching environments for defaults::python==3.6.4=hc3d631a_1: done
location: /home/maba/code/conda/conda-4.4
package matches:
  - defaults::python-3.6.4-hc3d631a_1

```
(Not necessarily suggesting to use the wide layout (I also like the one from `pref.dist_str()` 😄), just pointing out.)

Reply:
I think what especially matters here is the format of the `--json` flag, both with and without `--envs`.  I believe we're good there, but I'll check.

I could maybe make the `--envs` human format wide column as well, but I think we'd want prefix location to be column 0, so if someone is text parsing the columnar output here (don't do it people!), the format is still going to be slightly different.

Reply:
I think either put prefix in the last column or do a columnar output per prefix and put the prefix atop (i.e., like in the current `--envs` output).
```
$ conda search defaults::python=3.6.4=hc3d631a_1
Searching environments for defaults::python==3.6.4=hc3d631a_1: done
Name                       Version                   Build  Location
python                     3.6.4                hc3d631a_1  /home/maba/code/conda/conda-4.4
python                     3.6.4                hc3d631a_1  /home/maba/code/conda/conda-4.4/envs/x
```
```
$ conda search defaults::python=3.6.4=hc3d631a_1 --envs
Searching environments for defaults::python==3.6.4=hc3d631a_1: done
Name                       Version                   Build  Channel
-------------------------------------------------------------------
Location: /home/maba/code/conda/conda-4.4
python                     3.6.4                hc3d631a_1  defaults
-------------------------------------------------------------------
Location: /home/maba/code/conda/conda-4.4/envs/x
python                     3.6.4                hc3d631a_1  defaults
```
```
$ conda search defaults::python=3.6.4=hc3d631a_1 --envs
Searching environments for defaults::python==3.6.4=hc3d631a_1: done
-------------------------------------------------------------------
Location: /home/maba/code/conda/conda-4.4
Name                       Version                   Build  Channel
python                     3.6.4                hc3d631a_1  defaults
-------------------------------------------------------------------
Location: /home/maba/code/conda/conda-4.4/envs/x
Name                       Version                   Build  Channel
python                     3.6.4                hc3d631a_1  defaults
```
Putting location in the first column would shift the other columns too much to the right. Placing it at the end would allow the most compact/consistent layout.

Reply:
Uh, I like having both "Channel" and "Location". Looks pretty
```
(base) $ conda search --envs python
Searching environments for python: done
# Name                  Version           Build  Channel              Location            
python                    3.6.4      hc3d631a_1  pkgs/main            /home/maba/code/conda/conda-4.4
python                    3.6.4      hc3d631a_0  pkgs/main            /home/maba/code/conda/conda-canary
python                    3.6.3               1  conda-forge          /home/maba/code/conda/conda-forge/envs/bioconda
python                    3.6.3               1  conda-forge          /home/maba/code/conda/conda-forge/envs/code
python                    3.6.4               0  conda-forge          /home/maba/code/conda/conda-forge/envs/numba
python                    3.6.4      hc3d631a_0  pkgs/main            /home/maba/code/conda/conda-git-4.4.x
python                   2.7.14     h1571d57_29  pkgs/main            /home/maba/code/conda/conda-py2
python                    3.6.4      hc3d631a_1  pkgs/main            /home/maba/code/conda/defaults
```